app/code/Magento/Eav/Model/Validator/Attribute/Code.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Eav\Model\Validator\Attribute;
+
+use Magento\Eav\Model\Entity\Attribute;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Validator\AbstractValidator;
+use Zend_Validate;
+
+/**
+ * Class Code
+ *
+ * Validation EAV attribute code
+ */
+class Code  extends AbstractValidator
+{
+    /**
+     * Validates the correctness of the attribute code
+     *
+     * @param string $attributeCode
+     * @return bool
+     * @throws LocalizedException
+     */
+    public function isValid($attributeCode): bool
```

Comment:
The method named `isValid` should return bool and should not throw an exception. Validation messages should be collected using `_addMessages` parent method. Appropriate exceptions should be thrown if needed by client classes